deeplearning4j-core/src/test/java/org/deeplearning4j/nn/layers/OutputLayerTest.java

```diff
@@ -441,4 +445,81 @@ public void testRnnOutputLayerIncEdgeCases() {
             assertArrayEquals(preoutRnn.shape(), new int[] {miniBatchSize, nOut, timeSeriesLength});
         }
     }
+
+
+    @Test
+    public void testCompareRnnOutputRnnLoss(){
+        Nd4j.getRandom().setSeed(12345);
+
+        int timeSeriesLength = 4;
+        int nIn = 5;
+        int layerSize = 6;
+        int nOut = 6;
+        int miniBatchSize = 3;
+
+        MultiLayerConfiguration conf1 =
+                new NeuralNetConfiguration.Builder().seed(12345L)
+                        .updater(new NoOp())
+                        .list()
+                        .layer(new LSTM.Builder().nIn(nIn).nOut(layerSize).activation(Activation.TANH)
+                                .weightInit(WeightInit.DISTRIBUTION).dist(new NormalDistribution(0, 1.0))
+                                .updater(new NoOp()).build())
+                        .layer(new DenseLayer.Builder().nIn(layerSize).nOut(nOut).activation(Activation.IDENTITY).build())
```

Comment:
@AlexDBlack I'm curious: going forward, do you think it would be possible to write something like this instead:

```
     .layer(new DenseLayer.Builder().nIn(layerSize).nOut(nOut).activation(Activation.SOFTMAX).build())
     .loss(new RnnLossLayer.Builder(LossFunction.MCXENT).build())
```
or even:
```
     .layer(new DenseLayer.Builder().nIn(layerSize).nOut(nOut).activation(Activation.SOFTMAX).build())
     .loss(LossFunction.MCXENT)
```
i.e. inferring loss layer type?

Sorry, I know this is vastly beyond the scope of this PR. Just want to discuss.

Reply:
btw, this test was very useful for understanding what you try to achieve here. 👍

Reply:
That's a good idea, open an issue? Not sure on the exact semantics/names, but something like that would be good.